openrr-base/src/robot_velocity_status.rs

```diff
@@ -0,0 +1,160 @@
+use arci::BaseVelocity;
+use parking_lot::Mutex;
+
+use crate::*;
+
+#[derive(Debug, Default)]
+pub struct RobotVelocityStatus {
+    velocity: Mutex<BaseVelocity>,
+    feedback_velocity: Mutex<BaseVelocity>,
+    max_velocity: BaseVelocity,
+    min_velocity: BaseVelocity,
+    max_acceleration: BaseAcceleration,
+    min_acceleration: BaseAcceleration,
+    last_sent_log: Mutex<BaseVelocityTimestamped>,
+}
+
+impl RobotVelocityStatus {
+    pub fn new(max_velocity: BaseVelocity, max_acceleration: BaseAcceleration) -> Self {
+        Self::new_as_asymmetric(
+            max_velocity,
+            max_velocity * (-1.),
+            max_acceleration,
+            max_acceleration * (-1.),
+        )
+    }
+
+    pub fn new_as_asymmetric(
+        max_velocity: BaseVelocity,
+        min_velocity: BaseVelocity,
+        max_acceleration: BaseAcceleration,
+        min_acceleration: BaseAcceleration,
+    ) -> Self {
+        Self {
+            velocity: Mutex::new(BaseVelocity::default()),
+            feedback_velocity: Mutex::new(BaseVelocity::default()),
+            max_velocity,
+            min_velocity,
+            max_acceleration,
+            min_acceleration,
+            last_sent_log: Mutex::new(BaseVelocityTimestamped::default()),
+        }
+    }
+
+    pub fn get_velocity(&self) -> BaseVelocity {
```

Comment:
I feel better to remove `get_`

```suggestion
    pub fn velocity(&self) -> BaseVelocity {
```